src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -17,19 +18,40 @@
 @Singleton
 public class BrAPITrialService {
 
-    @Property(name = "brapi.server.reference-source")
-    private String referenceSource;
-
     private final BrAPITrialDAO trialDAO;
-    private final ProgramService programService;
+    private final BrAPIObservationUnitDAO ouDAO;
 
     @Inject
-    public BrAPITrialService(ProgramService programService, BrAPITrialDAO trialDAO) {
-        this.programService = programService;
+    public BrAPITrialService( BrAPITrialDAO trialDAO,BrAPIObservationUnitDAO ouDAO) {
         this.trialDAO = trialDAO;
+        this.ouDAO = ouDAO;
     }
 
     public List<BrAPITrial> getExperiments(UUID programId) throws ApiException, DoesNotExistException {
         return trialDAO.getTrials(programId);
     }
+
+    public HashMap<String, Object> getTrialDataByUUID(UUID programId, UUID trialId, boolean stats) throws DoesNotExistException {
```

Comment:
I think this should return a `BrAPITrial` object instead of a map, and I think the stats info can go in the `additionalInfo` field of the `BrAPITrial` object.

Reply:
done